src/components/NcDialog/NcDialog.vue

```diff
@@ -223,7 +223,7 @@ import { createElementId } from '../../utils/createElementId.ts'
 
 type NcDialogButtonProps = ComponentProps<typeof NcDialogButton>
 
-const props = withDefaults(defineProps<{
+export type NcDialogProps = {
```

Comment:
When defining the props type for export, you cannot use `defineModel`. `open` is missing in the type.

Instead, you need to define `open` and `update:open` explicitly, and then replace `defineModel` compile macros with `useModel` vue function.